src/commands/client.ts

```diff
@@ -404,6 +415,25 @@ export class CommandClient extends Client implements CommandClientOptions {
       }
     }
 
+    const userCooldowns = this.lastUsed.get(msg.author.id) ?? {
+      global: 0,
+      commands: {}
+    }
+    const commandCanBeUsedAt =
+      (userCooldowns.commands[command.name] ?? 0) +
+      ((command.cooldown ?? 0) as number)
+    const globalCooldown = userCooldowns.global + this.cooldown
```

Comment:
shouldn't one of them override the cooldown or am I reading the code wrong?

Reply:
but it adds the command cooldown with the global cooldown...

Reply:
my 2 cents:
global cooldown should prevent any user from running command `x` before the cooldown expires.

For example: resource heavy commands and/or for minigames like having a command that can only be redeemed by one person every `y` hours.

```ts
/**
* Defines a cooldown in miliseconds, preventing the same user from running the command within the cooldown period
**/
command.cooldown?: number
/**
* Defines a global cooldown in miliseconds, preventing any user from running the command within the cooldown period
**/
command.globalCooldown?:number
```

Reply:
> my 2 cents: global cooldown should prevent any user from running command `x` before the cooldown expires.
> 
> For example: resource heavy commands and/or for minigames like having a command that can only be redeemed by one person every `y` hours.
> 
> ```ts
> /**
> * Defines a cooldown in miliseconds, preventing the same user from running the command within the cooldown period
> **/
> command.cooldown?: number
> /**
> * Defines a global cooldown in miliseconds, preventing any user from running the command within the cooldown period
> **/
> command.globalCooldown?:number
> ```

I've added the global cooldowns and also renamed the previous cooldown to hopefully reduce confusion in regards to what it applies to.